### PR TITLE
Integrating with travis.

### PR DESCRIPTION
**Pull request by code-twister**
Adding maven deployment mechanism.

updated README.md with build status
removed old .ci.yml

### CODE REVIEW

File: .buildscript/deploy_snapshot.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  echo "Skipping snapshot deployment: wrong JDK. Expected '$JDK' but was '$TRAVIS_JDK_VERSION'."
+elif [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
+  echo "Skipping snapshot deployment: was pull request."
+elif [ "$TRAVIS_BRANCH" != "$BRANCH" ]; then
+  echo "Skipping snapshot deployment: wrong branch. Expected '$BRANCH' but was '$TRAVIS_BRANCH'."
+else
+  echo "Deploying snapshot..."
+  ./gradlew uploadArchives
+  echo "Snapshot deployed!"
+fi
```
Comment (stephanenicolas) [inline .buildscript/deploy_snapshot.sh]: empty line is missing.

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+### version 1.0.2 (TBD)
+
+
+
+### version 1.0.1 (July 24th, 2017)
+
+* initial release
```
Comment (stephanenicolas) [inline CHANGELOG.md]: can you add "on maven central."

File: README.md [hunk 3]
```diff
@@ -46,3 +49,7 @@ apt {
 ### Future plans
 
 We will add other useful checks related to dependency injection.
+
+### Credits
+
+ * Stephane Nicolas (https://github.com/stephanenicolas) - helped with code review and advice on the first version of the library
```
Comment (stephanenicolas) [inline README.md]: add a final period

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+cache:
+  directories:
+    - $HOME/.m2
+
+script:
+  - ./gradlew clean build
+
+after_success:
+#  - ./gradlew jacocoTestReport coveralls
+#  - .buildscript/deploy_snapshot.sh
```
Comment (stephanenicolas) [inline .travis.yml]: final empty line.

File: compiler/gradle.properties [hunk 5]
```diff
@@ -13,8 +13,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-
 POM_ARTIFACT_ID=compiler
 POM_NAME=DI Checks Compiler
-POM_DESCRIPTION='Annotation Processors for dependency injection checks'
```
Comment (stephanenicolas) [inline compiler/gradle.properties]: we should keep this line.

File: gradle/gradle-mvn-push.gradle [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+        if (project.getPlugins().hasPlugin('com.android.application') ||
+                project.getPlugins().hasPlugin('com.android.library')) {
+            archives androidSourcesJar
+            archives androidJavadocsJar
+        } else {
+            archives sourcesJar
+            archives javadocJar
+        }
+    }
+}
```
Comment (stephanenicolas) [inline gradle/gradle-mvn-push.gradle]: empty line

**Comment by stephanenicolas** [comment]
Let's first have travis working, and then, release.

File: compiler/gradle.properties (hunk 5 above)
Comment (code-twister) [inline compiler/gradle.properties]: it's duplicated in the parent gradle.properties

**Pull request merged**